src/os/linux/etcpal/os_socket.c

```diff
@@ -822,10 +822,10 @@ void events_epoll_to_etcpal(const struct epoll_event* epoll_evt, const EtcPalPol
     *events_out |= (ETCPAL_POLL_ERR);
 }
 
-int poll_socket_compare(const EtcPalRbTree* tree, const EtcPalRbNode* node_a, const EtcPalRbNode* node_b)
+int poll_socket_compare(const EtcPalRbTree* tree, const void* value_a, const void* value_b)
 {
-  EtcPalPollSocket* a = (EtcPalPollSocket*)node_a->value;
-  EtcPalPollSocket* b = (EtcPalPollSocket*)node_b->value;
+  EtcPalPollSocket* a = (EtcPalPollSocket*)value_a;
```

Comment:
Shouldn't this be `const EtcPalPollSocket* a = (const EtcPalPollSocket*) value_a;`  The c-style cast is removing constness.  Assume this comment is on the next line and all the os_socket.c variants as well.

Reply:
In C, all casts are C-style casts 😄  but you're right. Will fix